test/framework/integration_framework/integration_test_framework.hpp

```diff
@@ -113,79 +120,45 @@ namespace integration_framework {
     }
     // fetch status of transaction
-    iroha::model::TransactionResponse::Status status;
+    std::shared_ptr<shared_model::proto::TransactionResponse> status;
     {
-      iroha::protocol::TxStatus proto_status;
       iroha::protocol::TxStatusRequest request;
-      request.set_tx_hash(iroha::hash(tx).to_string());
+      request.set_tx_hash(shared_model::crypto::toBinaryString(tx.hash()));
       iroha::protocol::ToriiResponse response;
       iroha_instance_->getIrohaInstance()->getCommandService()->StatusAsync(
           request, response);
-      proto_status = response.tx_status();
-
-      switch (proto_status) {
-        case iroha::protocol::TxStatus::STATELESS_VALIDATION_FAILED:
-          status =
-              iroha::model::TransactionResponse::STATELESS_VALIDATION_FAILED;
-          break;
-        case iroha::protocol::TxStatus::STATELESS_VALIDATION_SUCCESS:
-          status =
-              iroha::model::TransactionResponse::STATELESS_VALIDATION_SUCCESS;
-          break;
-        case iroha::protocol::TxStatus::STATEFUL_VALIDATION_FAILED:
-          status =
-              iroha::model::TransactionResponse::STATEFUL_VALIDATION_FAILED;
-          break;
-        case iroha::protocol::TxStatus::STATEFUL_VALIDATION_SUCCESS:
-          status =
-              iroha::model::TransactionResponse::STATEFUL_VALIDATION_SUCCESS;
-          break;
-        case iroha::protocol::TxStatus::COMMITTED:
-          status = iroha::model::TransactionResponse::COMMITTED;
-          break;
-        case iroha::protocol::TxStatus::IN_PROGRESS:
-          status = iroha::model::TransactionResponse::IN_PROGRESS;
-          break;
-        case iroha::protocol::TxStatus::NOT_RECEIVED:
-        default:
-          status = iroha::model::TransactionResponse::NOT_RECEIVED;
-          break;
-      }
+      status =
+          std::make_shared<shared_model::proto::TransactionResponse>(response);
     }
     // check validation function
-    validation(status);
+    validation(*status);
     return *this;
   }
 
   template <typename Lambda>
   IntegrationTestFramework &IntegrationTestFramework::sendQuery(
-      const iroha::model::Query &qry, Lambda validation) {
+      shared_model::proto::Query qry, Lambda validation) {
     using iroha::operator|;
     log_->info("send query");
-    // serialize without calling destructor on passed reference
-    auto pb_qry = iroha::model::converters::PbQueryFactory().serialize(
-        std::shared_ptr<const iroha::model::Query>(&qry, [](auto) {}));
-    // send
-    iroha::protocol::QueryResponse pb_response;
-    iroha_instance_->getIrohaInstance()->getQueryService()->FindAsync(
-        *pb_qry, pb_response);
-    // deserialize
-    auto response =
-        iroha::model::converters::PbQueryResponseFactory().deserialize(
-            pb_response);
-    // check validation function
-    validation(**response);
+
+    std::shared_ptr<shared_model::proto::QueryResponse> query_response;
+    {
```

Comment:
Why is this code block required?

Reply:
Similar to the sendTx, so I though these bunch of routines can be extracted to separate methods. What do you think?